### PR TITLE
Icu 10178 use hds table in groups list view

### PR DESCRIPTION
**Pull request by NatashaKab**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10178)

## Description

:technologist: [Admin preview](https://boundary-ui-git-icu-10178-use-hds-table-in-gro-633926-hashicorp.vercel.app/scopes/global/groups)

Use HDS table in groups list view

### CODE REVIEW

File: ui/admin/app/templates/scopes/scope/groups/index.hbs [hunk 1]
```diff
@@ -31,42 +31,42 @@
...
-                >
-                  <code>{{group.id}}</code>
-                </Copyable>
-              </row.cell>
-            </body.row>
-          {{/each}}
-        </table.body>
-      </Rose::Table>
+      <Hds::Table
+        class='groups-list-table'
```
Comment (DhariniJeeva) [inline ui/admin/app/templates/scopes/scope/groups/index.hbs:35]: do we need a classname for this?

Comment (NatashaKab) [reply ui/admin/app/templates/scopes/scope/groups/index.hbs:35]: I don't know I just added it since the reference I used had that.

Comment (DhariniJeeva) [reply ui/admin/app/templates/scopes/scope/groups/index.hbs:35]: looks like it's not being used anywhere. can we remove this class and also the `users-list-table` from the users list?

File: ui/admin/app/templates/scopes/scope/groups/index.hbs [hunk 2]
```diff
@@ -31,42 +31,41 @@
...
+              {{#if (can 'read group' B.data)}}
+                <LinkTo @route='scopes.scope.groups.group' @model={{B.data.id}}>
+                  {{B.data.displayName}}
+                </LinkTo>
+              {{else}}
+                {{B.data.displayName}}
+              {{/if}}
+              {{#if B.data.accountName}}
+                <br /><Hds::Badge @text={{B.data.accountName}} />
+              {{/if}}
```
Comment (DhariniJeeva) [inline ui/admin/app/templates/scopes/scope/groups/index.hbs:54]: This is not needed

**Review by DhariniJeeva — CHANGES_REQUESTED** [review]

**Review by ZedLi — APPROVED** [review]
Looks good!

**Review by lisbet-alvarez — APPROVED** [review]

**Review by DhariniJeeva — APPROVED** [review]
great job!

**Review by cameronperera — APPROVED** [review]
LGTM

**Pull request merged**